### ISSUE TITLE
Query loop: Non-inherited queries must also update the document title (potential impact on SEO)

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#55489]
### Description

Similar to https://github.com/WordPress/gutenberg/issues/51579

When implementing an 'in page' navigation à la 'single page app', a few things must always be implemented. Amongst them, a routing mechanism, integration with the browser history API, and announcing navigation occurred to assistive technology.

One more fundamental aspect is that the document title must be updated.
If it isn't updated:
- _All entries_ in the browser history will have the same name.
  - This makes the browser history UI unusable, since there is no way for users to distinguish which page is which.
  - Edge caseL It makes pointless adding pagination pages to the bookmarks as all the saved bookmark will have the same name.
  - **It impacts SEO** as the paginated pages will be indexed but they have the same document title.

The 'classic' pagiination always adds 'page 2',  'page 3', etc. to the document title:

![classic pagination](https://github.com/WordPress/gutenberg/assets/1682452/04b30f45-f00b-4663-a9f8-8186f31c705f)

The 'Enhanced' pagination does not:

![enhanced pagination](https://github.com/WordPress/gutenberg/assets/1682452/2cc91142-3bba-479f-8ebf-b33e2eb33e6b)

### Step-by-step reproduction instructions

- Go to the wp Dashboard > Settings > Reading settings
- Set 'Blog pages show at most' to 5, just to have more paginated pages
- View the fron end and to to an archive page, e.g. an yearly archive. I tested with: http://localhost:8889/2012/
- Go back and forth the pagianted pages a few times.
- Observe the document title changes adding the page number.
- Click and hold the browser Back or Forward buttons.
- Observe each entry can be easily distinguished because of the page number.
- Checl also the Full history page of your browser.
- Edit the archive template in use in the editor.
- Select the Query loop block and enable 'Enhanced pagination'.
- Repeat the steps above.
- Observe all the entries in the history have the same name.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Worth also mentioning that a Query Loop block with 'Enhanced pagination' can be added also to a normal post or page.

In this case, _the post or page actually become an archive page_ with paginated pages. Still, the document title will not change and will always be the same across all paginated pages.

Tje URL will change, e.g.:
```
https://my-site.org/my-awesome-post/?query-1-page=1
https://my-site.org/my-awesome-post/?query-1-page=2
https://my-site.org/my-awesome-post/?query-1-page=3
```

The document title will stay the same for all paginated pages, e.g.:
```
<title>My awesome post – My site</title>
```

**Comment by fabiankaegy** [comment]
@afercia I think your last comment conflates two things. It has always been the case that the pagination added on a post / page that uses the query param based pagination doesn't update the page title. 

So I think there are two issues here. One is a long standing one that you are now finding, and the other is a new introduction with 6.4 and the enhanced pagination.

The new issue is that enhanced pagination on archive templates where you use the inherit option doesn't update the document title.

**Comment by afercia** [comment]
> It has always been the case that the pagination added on a post / page that uses the query param based pagination doesn't update the page title.

@fabiankaegy yes that is possible but I would say it's an edge case. Worth improving it though.

The main issue here is about the archive pages. With SEO impact still to fully evaluate.

**Comment by afercia** [comment]
> It has always been the case that the pagination added on a post / page that uses the query param based pagination doesn't update the page title.

I see a substantial difference between the situation with classic themes and block-based themes though.

Before:
adding some archive-like pagination to post / page and attempt to uses the query param based pagination was essentially a task for theme  plugin developers. As developers, they had the technical skills to take care of the document title, if they wanted.

Now:
Users can and a Query Loop block and the Pagination block everywhere. As such, WordPress should make sure that it works correctly everywhere, updating the document title when necessary.

**Comment by fabiankaegy** [comment]
> I see a substantial difference between the situation with classic themes and block-based themes though.
> 
> Before: adding some archive-like pagination to post / page and attempt to uses the query param based pagination was essentially a task for theme plugin developers. As developers, they had the technical skills to take care of the document title, if they wanted.
> 
> Now: Users can and a Query Loop block and the Pagination block everywhere. As such, WordPress should make sure that it works correctly everywhere, updating the document title when necessary.

I don't think I understand where you are drawing the line here. Regardless of whether you are in a block theme or in a classic theme that uses the block editor. You can insert a query block on any post / page in either. Which always has the query parameter based pagination which doesn't update the title. 

So the issue is the same for all kinds of themes. This issue is fundamental to the Query block.

And the addition of "enhanced pagination" doesn't change anything about that.


The only difference with block based themes is that you can now also use the query block on archive pages and therefore use that actual `/page/` based pagination. Before adding the "enhanced pagination" feature this did update the title correct. But now with the enhanced pagination the title is not updated correctly when the `inherit` option is selected.

**Comment by afercia** [comment]
> I don't think I understand where you are drawing the line here.

When I mentioned 'classic' themes and block-based themes, I meant 'classic' as in: the traditional themes where pagination is managed by WordPres stemplating functions and users can't mess with the Query Loop.

> Before adding the "enhanced pagination" feature this did update the title correct. But now with the enhanced pagination the title is not updated correctly when the `inherit` option is selected.

I think it doesn't update correctly also wjen `inherit` is disabled annd a custom query runs. Regardless, 'enhanced pagination' breaks a native browsers feature. Or, better, it males it unusable which is something that seems less than ideal to me. 

As I mentioned, there's also potential SEO concerns here still to be fully evaluated.

**Comment by cbravobernal** [comment]
Would be the enhanced pagination issue solved with this PR?

https://github.com/WordPress/gutenberg/pull/55446

**Comment by afercia** [comment]
https://github.com/WordPress/gutenberg/pull/55446 solves the accessibility and usability issue.
I'm not sure it fully solves the potential SEO issue. I'd like to hear thoughts from some SEO experts here.

**Comment by cbravobernal** [comment]
I'm not a SEO expert, but I guess when robots access different pages, they do a navigation per page, instead of clicking on the different pagination links. In that case all pages have their own urls and can be crawled, so should not be a problem.

**Comment by afercia** [comment]
I'm not an SEO expert so I'm not going to make assumptions about things I'm not 100% familiar with.
Just noting that it's not only about crawling, It's also about the Search results.

Anyways, it is important to take into consideration WordPress core has a few filters in place in the code that generates the document title. With Enhanced pagination enabled, these filters will keep returning the value generated via PHP on first page load as expected, but the value will not change when clicking the Enhanced pagination links. Basically these filters won't work and there's no equivalent implementation with Enhanced pagination enabled.
Plugins do use these filters to handle the title, description, opengraph tags and schemas. Also, the lack of the pagination related query_var(s) does have an impact and I'm not sure it has been fully evaluated.

Right now, while the document title is updated with the page number, other tags generated by plugins don't get the correct page number as they expect the WP filters to just work. 

I'd think that at the very least the WordPress 6.4 Field Guid should put this problem in big relevance.
Ideally, an equivalent JS implementation to the core filters should have been provided.

![Screenshot 2023-10-26 at 14 49 55](https://github.com/WordPress/gutenberg/assets/1682452/f6c8589f-bc14-437f-ada1-052e8d21961d)


Cc @enricobattocchi

**Comment by luisherranz** [comment]
I have renamed the title since the problem with enhanced pagination was solved in [this pull request](https://github.com/WordPress/gutenberg/pull/55446), and only the issue with non-inherited queries remains, which does not update the title even with enhanced pagination disabled.

Moreover, if non-inherited queries modify the page title someday, it will work by default with enhanced pagination because that's just updating the title with whatever is inside the `<title>` tag.
